client.go

```diff
@@ -6,37 +6,47 @@ import (
 	"errors"
 	"io/ioutil"
 	"net/http"
+	"net/url"
 	"strings"
 	"time"
 )
 
+// Client executes JSON RPC calls to remote servers.
 type Client struct {
-	Url     string
+	URL     string
 	Timeout int
+
+	proxyAddress string
 }
 
+// NewClient returns a newly istantiated Client pointing to the given url.
 func NewClient(url string) *Client {
-	client := new(Client)
-
-	client.Url = url
-	client.Timeout = DEFAULT_TIMEOUT
-
-	return client
+	return &Client{
+		URL:     url,
+		Timeout: defaultTimeout,
+	}
 }
 
-func (this *Client) sendJsonRequest(jsonRequest []byte) ([]byte, error) {
+func (c *Client) sendJSONRequest(jsonRequest []byte) ([]byte, error) {
 	var jsonResponse []byte
 
-	httpClient := &http.Client{
-		Timeout: time.Duration(time.Duration(this.Timeout) * time.Second),
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify: true,
-			},
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
 		},
+		Proxy: http.ProxyFromEnvironment,
+	}
+
+	if proxyURL, err := url.Parse(c.proxyAddress); c.proxyAddress != "" && err == nil {
+		transport.Proxy = http.ProxyURL(proxyURL)
 	}
 
-	httpRequest, err := http.NewRequest("POST", this.Url, strings.NewReader(string(jsonRequest)))
+	httpClient := &http.Client{
```

Comment:
Questo client HTTP possiamo metterlo come attributo del Client in modo da non istanziarlo ogni volta.